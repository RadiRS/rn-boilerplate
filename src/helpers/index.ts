export * from './test-helpers/render-helper';
export * from './mask/mask-helpers';
export * from './utils';
