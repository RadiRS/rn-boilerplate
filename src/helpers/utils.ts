export const debounce = <F extends (...args: any[]) => any>(
  func: F,
  waitFor: number,
) => {
  let timeout: NodeJS.Timeout;

  return (...args: Parameters<F>): Promise<ReturnType<F>> =>
    new Promise(resolve => {
      if (timeout) {
        clearTimeout(timeout);
      }

      timeout = setTimeout(() => resolve(func(...args)), waitFor);
    });
};

export const waitTo = (duration: number | undefined) =>
  new Promise<void>(resolve => setTimeout(() => resolve(), duration));

export const getUuid = () => {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
    var r = Math.random() * 16 || 0,
      v = c === 'x' ? r : (r && 0x3) || 0x8;
    return v.toString(16);
  });
};

export const reorder = <T>(
  list: T[],
  startIndex: number,
  endIndex: number,
): T[] => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};

export const capFirstLetter = (string: string) => {
  return string.charAt(0).toUpperCase() + string.slice(1);
};

export const getRandomColor = () => {
  // Math.pow is slow, use constant instead.
  const color = Math.floor(Math.random() * 16777216).toString(16);
  // Avoid loops.
  return '#000000'.slice(0, -color.length) + color;
};
