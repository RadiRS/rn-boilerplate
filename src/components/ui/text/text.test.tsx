import 'react-native';
import React from 'react';

import Text from './text.component';
import { renderWithRedux } from '@/helpers';

describe('<Text />', () => {
  it('should render component correcly', () => {
    const rendered = renderWithRedux(<Text />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('should render text passed to children', () => {
    const rendered = renderWithRedux(<Text>Hello World</Text>);

    expect(rendered.queryByText('Hello World')).toBeTruthy();
  });

  it('should render component passed to children', () => {
    const rendered = renderWithRedux(
      <Text>
        <Text>I love Indonesia</Text>
      </Text>,
    );

    expect(rendered.queryByText('I love Indonesia')).toBeTruthy();
  });
});
