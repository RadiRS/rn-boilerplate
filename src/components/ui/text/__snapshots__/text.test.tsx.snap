// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render component correcly 1`] = `
<Text
  style={
    [
      {
        "color": "#212529",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 16,
        "textAlign": "left",
      },
      undefined,
    ]
  }
/>
`;
